insights/parsers/openshift_hosts.py

```diff
@@ -0,0 +1,117 @@
+        master.ose35.com  openshift_public_ip=192.66.208.202 openshift_ip=192.66.208.202 openshift_public_hostname=master.ose35.com openshift_hostname=master.ose35.com connect_to=master.ose35.com ansible_connection=local
+        [etcd]
+        master.ose35.com  openshift_public_ip=192.66.208.202 openshift_ip=192.66.208.202 openshift_public_hostname=master.ose35.com openshift_hostname=master.ose35.com connect_to=master.ose35.com ansible_connection=local
+
+    Examples:
+        >>> type(host_info)
+        <class 'insights.parsers.openshift_hosts.OpenShiftHosts'>
+        >>> host_info["OSEv3:children"]
+        ["nodes","nfs","masters","etcd"]
+        >>> host_info["nodes"]["master.ose35.com"]["openshift_public_ip"]
+        "192.66.208.202"
+        >>> host_info["nodes"]["master.ose35.com"]["openshift_node_labels"]
+        "{'region': 'infra'}"
+        >>> host_info.has_node("node1.ose35.com")
+        True
+        >>> host_info.has_var("openshift_use_crio")
+        False
+
+    """
+
+    def parse_content(self, content):
+        dict_all = {}
+        sub_section = ""
+        for line in get_active_lines(content):
+            line = line.strip()
+            if line.startswith("[") and line.endswith("]"):
+                sub_section = line[1:-1].strip()
+                if "OSEv3:children" in sub_section:
+                    dict_all[sub_section] = []
+                else:
+                    dict_all[sub_section] = {}
+                continue
+            else:
+                # Parser section [OSEv3:children]
+                if "OSEv3:children" in sub_section:
+                    dict_all[sub_section].append(line.strip())
+                # Parser section [OSEv3:vars]
+                elif "OSEv3:vars" in sub_section:
+                    key, value = line.split("=")
+                    dict_all[sub_section][key.strip()] = value.strip()
+                else:
+                    nodeinfo = line.strip().split()
+                    nodename = nodeinfo[0]
+                    dict_all[sub_section][nodename] = {}
+                    former_key = ""
+                    for info in nodeinfo[1:]:
+                        # There is key-value like "openshift_node_labels="{'region': 'infra'}"", in the nodeinfo list,
+                        # it will be splitted as [openshift_node_labels="{'region':, 'infra'}"]. In this for loop, second
+                        # value would be added to former value, and use same key "openshift_node_labels".
+                        if "=" not in info and "}" in info:
+                            dict_all[sub_section][nodename][former_key] = dict_all[sub_section][nodename][former_key] + info.strip()
+                        else:
+                            key, value = info.split("=")
+                            dict_all[sub_section][nodename][key.strip()] = value.strip()
+                            if "{" in info and ":" in info:
+                                former_key = key.strip()
```

Comment:
@wushiqinlou is [this reference](https://docs.openshift.com/container-platform/3.5/install_config/install/advanced_install.html) applicable to this file?  If so then I don't believe this parser will work as there are multiple example showing multiple key/value pairs in the `{}` and also settings split across lines.